Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -609,6 +609,379 @@ Start Safari on the iOS device. Run the above Python script on your connected ho
+```
+
+In fact, the string can be already found in memory in a rw- (read and write) region.
+
+Additionally, you can search for occurrences of the [wide version of the string](https://en.wikipedia.org/wiki/Wide_character "Wide character") (`\/w`) and, again, check their memory regions:
+
+> This time we run the `\dm.` command for all `@@` hits matching the glob `hit5_*`.
+
+```bash
+[0x00000000]> \/w owasp-mstg
+Searching 20 bytes: 6f 00 77 00 61 00 73 00 70 00 2d 00 6d 00 73 00 74 00 67 00
+Searching 20 bytes in [0x0000000100708000-0x000000010096c000]
+...
+hits: 2
+0x1020d1280 hit5_0 6f0077006100730070002d006d00730074006700
+0x1030c9c85 hit5_1 6f0077006100730070002d006d00730074006700
+
+[0x00000000]> \dm.@@ hit5_*
+0x0000000102000000 - 0x0000000102100000 rw-
+0x0000000103084000 - 0x00000001030cc000 rw-
+```
+
+They are in a different rw- region. Note that searching for the wide versions of strings is sometimes the only way to find them as you'll see in the following section.
+
+In-memory search can be very useful to quickly know if certain data is located in the main app binary, inside a shared library or in another region. You may also use it to test the behavior of the app regarding how the data is kept in memory. For instance, you could continue the previous example, this time clicking on Login and searching again for occurrences of the data. Also, you may check if you still can find those strings in memory after the login is completed to verify if this _sensitive data_ is wiped from memory after its use.
+
+###### Memory Dump
+
+Wether you are using a jailbroken device with frida-server installed or a non-jailbroken device, you can dump the app's process memory with [objection](https://github.com/sensepost/objection "Objection") and [Fridump](https://github.com/Nightbringer21/fridump "Fridump"). To take advantage of these tools on a non-jailbroken device, the iOS app must be repackaged with `FridaGadget.dylib` and re-signed. A detailed explanation of this process is in the section "[Dynamic Analysis on Non-Jailbroken Devices](#dynamic-analysis-on-non-jailbroken-devices "Dynamic Analysis on Non-Jailbroken Devices").
+
+With objection it is possible to dump all memory of the running process on the device by using the command `memory dump all`.
+
+```shell
+$ objection explore
+
+iPhone on (iPhone: 10.3.1) [usb] # memory dump all /Users/foo/memory_iOS/memory
+Dumping 768.0 KiB from base: 0x1ad200000  [####################################]  100%
+Memory dumped to file: /Users/foo/memory_iOS/memory
+```
+
+Alternatively you can use Fridump.
+
+> The original version of Fridump is no longer maintained, and the tool works only with Python 2. The latest Python version (3.x) should be used for Frida, so Fridump doesn't work out of the box.
+>
+> If you're getting the following error message despite your iOS device being connected via USB, checkout [Fridump with the fix for Python 3](https://github.com/sushi2k/fridump "Fridump for Python3").
```

Comment:
so how can we make this work?

Reply:
Actually I didn't have any problems. It just worked following the standard instructions. I'd remove these paragraphs (someone put them there before me).